File: src/lib-common/pad-r.js
function pad_r(x,y,z)
{
 check(is_cool,x)

 if(is_uint(x))
 {
  const s=to_str(x)
  
  if(is_undef(y))
  {
   if(is_undef(z))
    return pad_r(s,"0",3)

   return pad_r(s,"0",z)
  }
  
  return pad_r(s,y,z)
 }

 check(is_str,x)
 check(is_str,y)
 check(is_uint,z)

 return x.padEnd(z,y)
}

File: src/lib-common/pad-r.cs
fn pad_r x:cool padding length
 if is_uint x
  let s to_json x

  if is_undef padding
   if is_undef length
    ret pad_r s "0" 3

   ret pad_r s "0" length
  end

  ret pad_r s padding length
 end

 check is_str x
 check is_str padding
 check is_uint length

 ret x.padEnd length padding
end
